apm-agent-core/src/main/java/co/elastic/apm/agent/bci/ElasticApmAgent.java

```diff
@@ -448,7 +448,7 @@ private static void validateAdvice(Class<?> adviceClass) {
         }
         for (MethodDescription.InDefinedShape exitAdvice : typeDescription.getDeclaredMethods().filter(isStatic().and(isAnnotatedWith(Advice.OnMethodExit.class)))) {
             validateAdviceReturnAndParameterTypes(exitAdvice);
-            if (exitAdvice.getReturnType().getTypeName().startsWith("co.elastic.apm")) {
+            if (exitAdvice.getReturnType().asRawType().getTypeName().startsWith("co.elastic.apm")) {
```

Comment:
This is to avoid that the type gets resolved (leading to exceptions). That part of the code will change once we merge in the changes you made for WebFlux. So adding a comment would make merging trickier.

Reply:
Thank you for taking care of making other's PRs easier to merge !!